pkg/util/iterutil/iterutil.go

```diff
@@ -28,15 +28,18 @@ var errStopIteration = errors.New("stop iteration")
 //
 // 	for i := range slice {
 // 		if err := f(i); err != nil {
-// 			if iterutil.Done() {
-// 				return nil
-// 			}
-// 			return err
+// 			return iterutil.Map(err)
 // 		}
 // 		// continue when nil error
 // 	}
 //
 func StopIteration() error { return errStopIteration }
 
-// Done tells if the error is ErrStopIteration, i.e., should the iteration stop.
-func Done(err error) bool { return errors.Is(err, errStopIteration) }
+// Map the nil if it is StopIteration, or keep the error otherwise
+func Map(err error) error {
```

Comment:
If we want iterutil to be more monadic shouldn't this be a `Filter` instead of a `Map`?

Reply:
I spent a little time thinking about how to do this, and couldn't come up with a great solution. If anything the original `Done` was much closer to `Filter`. The problem with it in this case, is trying to both exit a loop early as well as return either success or fail.

Let me know if you think I should look at a different alternative or just merge this one in.

Reply:
Your change is a strict improvement over what there was so I feel very good about it 👍